### PR TITLE
avro content type detection bug

### PR DESCRIPTION
**Pull request by famartinrh**
For now this PR just contains tests to verify the behavior we currently have

### CODE REVIEW

**Comment by famartinrh** [comment]
@jsenko @carlesarnal this is now ready for review

File: app/src/main/java/io/apicurio/registry/ccompat/store/RegistryStorageFacadeImpl.java [hunk 1]
```diff
@@ -91,9 +90,17 @@
...
-        final ContentHandle contentHandle = storage.getArtifactByContentId(globalId);
-        return FacadeConverter.convert(contentHandle, ArtifactTypeUtil.discoverType(contentHandle, null));
+    public SchemaInfo getSchemaById(int contentId) throws ArtifactNotFoundException, RegistryStorageException {
+        final ContentHandle contentHandle = storage.getArtifactByContentId(contentId);
+        List<ArtifactMetaDataDto> artifacts = storage.getArtifactVersionsByContentId(contentId);
+        if (artifacts == null || artifacts.isEmpty()) {
+            //the contentId points to an orphaned content
+            throw new ArtifactNotFoundException("ContentId: " + contentId);
+        }
+        //get the artifact type for the first artifact version associated with it
```
Comment (carlesarnal) [inline app/src/main/java/io/apicurio/registry/ccompat/store/RegistryStorageFacadeImpl.java:100]: True, but if you upload a, e.g., JSON and you say it's XML you're doing that at your own risk.

**Review by carlesarnal — APPROVED** [review]
LGTM. If I'm reading this correctly, the actual bug at the compatibility layer was caused by the `determineArtifactType` not discovering Avro. Nice!

File: app/src/main/java/io/apicurio/registry/ccompat/store/RegistryStorageFacadeImpl.java (hunk 1 above)
Comment (famartinrh) [reply app/src/main/java/io/apicurio/registry/ccompat/store/RegistryStorageFacadeImpl.java:100]: completely agree, just leaved that text for historical context

**Review by jsenko — APPROVED** [review]

**Pull request merged**